### PR TITLE
add ported tests from puppeteer for page.click

### PR DESCRIPTION
**Pull request by Granitosaurus**

### CODE REVIEW

File: tests/test_click.py [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+    await p.setViewport(dict(width=500, height=500))
+    await p.setContent('<div style="width:100px;height:2000px">spacer</div>')
+    await attachFrame(p, server.cross_process_server / '/input/button.html', 'button-test')
+    frame = p.frames[1]
+    await frame.evaluate(
+        'button',
+        "button => button.style.setProperty('position', 'fixed')"
+    )
+    await frame.click('button')
+    assert await frame.evaluate("() => window.result") == 'Clicked'
```
Comment (Mattwmaster58) [inline tests/test_click.py]: Looks like this is basically duplicated below, so I don't see why we wouldn't be able to remove it.

File: tests/test_click.py [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+
+
+@sync
+async def test_click_button_devicescalefactor(isolated_page, server):
+    p = isolated_page
+    await p.setViewport(dict(width=500, height=500, deviceScaleFactor=5))
```
Comment (Mattwmaster58) [inline tests/test_click.py]: I'd prefer a dict literal here
```suggestion
    await p.setViewport({'width': 500, 'height': 500, 'deviceScaleFactor': 5})
```

**Review by Mattwmaster58 — CHANGES_REQUESTED** [review]
Looks good for the most part. I would replace `'\(\) => ([\w\.]+)'` with `$1` though as it's more readable and functionally the same. (eg `() => window.CLICKED` gets replaced with `window.CLICKED`). Another thing is that you don't need the leading / when referring to a server destination. (ie `server / '/page.html'` can become `server / 'page.html'`, which IMO is more readable)

**Comment by Granitosaurus** [comment]
Ah I didn't know the arrow functions aren't necessary, good to see we can get rid a bit of this clutter! Updated with your proposed style changes.

**Pull request merged**